src/ompl/util/src/RandomNumbers.cpp

```diff
@@ -217,3 +229,260 @@ void ompl::RNG::eulerRPY(double value[3])
     value[1] = acos(1.0 - 2.0 * uni_()) - boost::math::constants::pi<double>() / 2.0;
     value[2] = boost::math::constants::pi<double>() * (-2.0 * uni_() + 1.0);
 }
+
+
+void ompl::RNG::uniformNormalVector(unsigned int n, double value[])
+{
+    //Construct the Boost distribution and generator. These would ideally be at the class level, but the spherical distribution requires dimension at construction
+    boost::uniform_on_sphere<> uniformSphereDist(n);
+    boost::variate_generator<boost::mt19937&, boost::uniform_on_sphere<> > normalizedVector(generator_, uniformSphereDist);
+
+    //A temporary vector for the BOOST return value
+    std::vector<double> rVector;
+
+    //Generate a random value, the variate_generator returns a std::vector
+    rVector = normalizedVector();
+
+    //Copy into the array, wish I knew of a better way than having used the temporary vector...
```

Comment:
There *is* a better way. You can the boost::numeric::ublas::vector and boost::numeric::rubles::shallow_array_adaptor. See ompl/src/ompl/base/src/ProjectionEvaluator.cpp to see an example.